"No backing RepositoryOperations configured for repository" in native image.

### Expected Behavior

_No response_

### Actual Behaviour

We use R2dbcRepository and on JDK8 all works fine. On native-image we catch an error
```
ERROR i.m.h.n.stream.HttpStreamsHandler - Error occurred writing stream response: No backing RepositoryOperations configured for repository. Check your configuration and try again

micronautguide    | io.micronaut.context.exceptions.ConfigurationException: No backing RepositoryOperations configured for repository. Check your configuration and try again

micronautguide    |  at io.micronaut.data.intercept.DataIntroductionAdvice.findInterceptor(DataIntroductionAdvice.java:124)

micronautguide    |  at io.micronaut.data.intercept.DataIntroductionAdvice.intercept(DataIntroductionAdvice.java:78)

micronautguide    |  at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:137)

micronautguide    |  at io.micronaut.validation.ValidatingInterceptor.intercept(ValidatingInterceptor.java:138)

micronautguide    |  at io.micronaut.aop.chain.MethodInterceptorChain.proceed(MethodInterceptorChain.java:137)

micronautguide    |  at example.micronaut.AuthorRepository$Intercepted.save(Unknown Source)

micronautguide    |  at example.micronaut.AuthorController.create$lambda-0(AuthorController.kt:29)

micronautguide    |  at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:125)

micronautguide    |  at reactor.core.publisher.MonoNext$NextSubscriber.onNext(MonoNext.java:82)

micronautguide    |  at reactor.core.publisher.MonoNext$NextSubscriber.onNext(MonoNext.java:82)

micronautguide    |  at io.micronaut.http.server.netty.binders.PublisherBodyBinder$1.doOnNext(PublisherBodyBinder.java:108)

micronautguide    |  at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onNext(CompletionAwareSubscriber.java:52)

micronautguide    |  at io.micronaut.http.server.netty.jackson.JsonContentProcessor$1.doOnNext(JsonContentProcessor.java:130)

micronautguide    |  at io.micronaut.http.server.netty.jackson.JsonContentProcessor$1.doOnNext(JsonContentProcessor.java:98)

micronautguide    |  at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onNext(CompletionAwareSubscriber.java:52)

micronautguide    |  at io.micronaut.jackson.parser.JacksonProcessor.publishNode(JacksonProcessor.java:211)

micronautguide    |  at io.micronaut.jackson.parser.JacksonProcessor.onUpstreamMessage(JacksonProcessor.java:178)

micronautguide    |  at io.micronaut.jackson.parser.JacksonProcessor.onUpstreamMessage(JacksonProcessor.java:49)

micronautguide    |  at io.micronaut.core.async.processor.SingleThreadedBufferingProcessor.doOnNext(SingleThreadedBufferingProcessor.java:56)

micronautguide    |  at io.micronaut.core.async.subscriber.SingleThreadedBufferingSubscriber.onNext(SingleThreadedBufferingSubscriber.java:91)

micronautguide    |  at io.micronaut.http.server.netty.jackson.JsonContentProcessor.onData(JsonContentProcessor.java:152)

micronautguide    |  at io.micronaut.http.server.netty.AbstractHttpContentProcessor.doOnNext(AbstractHttpContentProcessor.java:78)

micronautguide    |  at io.micronaut.http.server.netty.AbstractHttpContentProcessor.doOnNext(AbstractHttpContentProcessor.java:36)

micronautguide    |  at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onNext(CompletionAwareSubscriber.java:52)

micronautguide    |  at io.micronaut.http.netty.reactive.HandlerPublisher.publishMessage(HandlerPublisher.java:378)

micronautguide    |  at io.micronaut.http.netty.reactive.HandlerPublisher.access$600(HandlerPublisher.java:65)

micronautguide    |  at io.micronaut.http.netty.reactive.HandlerPublisher$ChannelSubscription.flushBuffer(HandlerPublisher.java:487)

micronautguide    |  at io.micronaut.http.netty.reactive.HandlerPublisher$ChannelSubscription.receivedDemand(HandlerPublisher.java:534)

micronautguide    |  at io.micronaut.http.netty.reactive.HandlerPublisher$ChannelSubscription.lambda$request$0(HandlerPublisher.java:474)

micronautguide    |  at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)

micronautguide    |  at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)

micronautguide    |  at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:497)

micronautguide    |  at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)

micronautguide    |  at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)

micronautguide    |  at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)

micronautguide    |  at java.lang.Thread.run(Thread.java:748)

micronautguide    |  at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:567)

micronautguide    |  at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:192)

micronautguide    | Caused by: io.micronaut.context.exceptions.NoSuchBeanException: No bean of type [io.micronaut.data.r2dbc.operations.R2dbcRepositoryOperations] exists. Make sure the bean is not disabled by bean requirements (enable trace logging for 'io.micronaut.context.condition' to check) and if the bean is enabled then ensure the class is declared a bean and annotation processing is enabled (for Java and Kotlin the micronaut-inject-java' dependency should be configured as an annotation processor).

micronautguide    |  at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2698)

micronautguide    |  at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:824)

micronautguide    |  at io.micronaut.data.intercept.DataIntroductionAdvice.findInterceptor(DataIntroductionAdvice.java:121)
```

### Steps To Reproduce

- Kotlin 1.5
- graalvm-ce:java8-21.2.0

### Environment Information

_No response_

### Example Application

_No response_

### Version

3.0.1

--- a/test-graalvm/h2/build.gradle.kts
+++ b/test-graalvm/h2/build.gradle.kts
@@ -1,8 +1,30 @@
 plugins {
     id("io.micronaut.build.internal.r2dbc-testproject")
+    id("org.jetbrains.kotlin.jvm")
+    id("org.jetbrains.kotlin.kapt")
+    id("org.jetbrains.kotlin.plugin.allopen")
+    id("io.micronaut.build.internal.r2dbc-kotlin")
+}
+
+allOpen {
+    annotation("io.micronaut.aop.Around")
+}
+
+tasks.named("internalStartTestResourcesService") {
+    enabled = false
+}
+
+tasks.named("startTestResourcesService") {
+    enabled = false
+}
+
+tasks.named("stopTestResourcesService") {
+    enabled = false
 }
 
 dependencies {
     runtimeOnly(libs.managed.r2dbc.h2)
     runtimeOnly(mnSql.h2)
+    kaptTest(mnData.micronaut.data.processor)
+    testImplementation(mnKotlin.micronaut.kotlin.runtime)
 }

--- a/test-graalvm/h2/src/test/kotlin/example/kotlin/KotlinOwnerRepository.kt
+++ b/test-graalvm/h2/src/test/kotlin/example/kotlin/KotlinOwnerRepository.kt
@@ -1,0 +1,12 @@
+package example.kotlin
+
+import io.micronaut.data.model.query.builder.sql.Dialect
+import io.micronaut.data.r2dbc.annotation.R2dbcRepository
+import io.micronaut.data.repository.reactive.ReactorCrudRepository
+import reactor.core.publisher.Mono
+import testgraalvm.domain.Owner
+
+@R2dbcRepository(dialect = Dialect.H2)
+interface KotlinOwnerRepository : ReactorCrudRepository<Owner, Long> {
+    fun findByName(name: String): Mono<Owner>
+}

--- a/test-graalvm/h2/src/test/kotlin/example/kotlin/KotlinRepositoryNativeImageTest.kt
+++ b/test-graalvm/h2/src/test/kotlin/example/kotlin/KotlinRepositoryNativeImageTest.kt
@@ -1,0 +1,35 @@
+package example.kotlin
+
+import io.micronaut.data.r2dbc.operations.R2dbcOperations
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Test
+import reactor.core.publisher.Mono
+import testgraalvm.domain.Owner
+import jakarta.inject.Inject
+
+@MicronautTest(transactional = false)
+class KotlinRepositoryNativeImageTest {
+
+    @Inject
+    lateinit var operations: R2dbcOperations
+
+    @Inject
+    lateinit var repository: KotlinOwnerRepository
+
+    @Test
+    fun kotlinRepositorySaveUsesRepositoryOperationsInNativeImage() {
+        val saved = Mono.from(operations.withTransaction {
+            repository.save(Owner("Wilma", 36))
+        }).block()
+
+        assertNotNull(saved)
+        assertNotNull(saved!!.id)
+        assertEquals("Wilma", saved.name)
+
+        val fetched = repository.findByName("Wilma").block()
+        assertNotNull(fetched)
+        assertEquals(saved.id, fetched!!.id)
+    }
+}

